ocamldoc/odoc_sig.ml

```diff
@@ -774,6 +830,7 @@ module Analyser =
             in
             let name = Name.parens_if_infix name_pre.txt in
             let subst_typ = Odoc_env.subst_type env type_expr in
+            let comment_opt = analyze_alerts comment_opt value_desc.Parsetree.pval_attributes in
```

Comment:
This is too early to merge alert and deprecation information: at this point, we have not yet collected the post-element documentation comments. Thus if we transfer the deprecation alerts to the deprecation info, when calling
```ocaml
 merge_infos v.val_info info_after_opt
```
later we end up merging the alert and the deprecation comment.
This is what is happening in the html test.

Reply:
Thanks for the debugging ! I think I've found a fix.